Scripts/AssemblyInjections/Unity.Entities/ComponentTypeDependencyExtension.cs

```diff
@@ -0,0 +1,171 @@
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Entities;
+using Unity.Jobs;
+using UnityEngine;
+
+/// <summary>
+/// A collection of extension methods to help calculate the scheduling dependencies on <see cref="ComponentType"/>s.
+/// </summary>
+[BurstCompatible]
+public static class ComponentTypeDependencyExtension
+{
+    private static UnsafeList<int> s_WriteTypeList_ScratchPad = new UnsafeList<int>(0, Allocator.Persistent);
+    private static UnsafeList<int> s_ReadTypeList_ScratchPad = new UnsafeList<int>(0, Allocator.Persistent);
```

Comment:
You might get the Memory leak error here because nothing ever disposes the collection before Unity quits.
I've been using the following pattern for this:

```csharp
[RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
        private static void Init()
        {
              if (s_WriteTypeList_ScratchPad.IsCreated)
              {
                     s_WriteTypeList.Dispose();
              }
              s_WriteTypeList_ScratchPad = new UnsafeList<int>(0, Allocator.Persistent);
          }
```

Reply:
Not sure I follow. If the domain gets reloaded the sandbox of memory gets reset right?
Otherwise, the static collections won't get re-instantiated anyway.

Your pattern just seems to reset the collection(s) between play sessions without code changes. I guess that does help with deterministic behaviour though

Reply:
> If the domain gets reloaded the sandbox of memory gets reset right?

Yes, but if you don't dispose your native collections, then Unity detects that and squawks at you because you didn't dispose before the sandbox was reset.

Reply:
ah, ok. will make the edit

Reply:
FYI this error doesn't trigger for `UnsafeList`. I added the change anyway for more per-run behaviour.

Reply:
Done